library/Icingadb/Web/Controller.php

```diff
@@ -327,12 +328,13 @@ public function handleSearchRequest(Query $query)
      * @param Query $query
      * @param string $search
      * @param Filter\Any $filter
+     * @param array $columns
      *
      * @return void
      */
-    protected function prepareSearchFilter(Query $query, string $search, Filter\Any $filter)
+    protected function prepareSearchFilter(Query $query, string $search, Filter\Any $filter, array $columns)
     {
-        foreach ($query->getModel()->getSearchColumns() as $column) {
+        foreach ($columns as $column) {
```

Comment:
`$columns` may be empty. If one doesn't pass them, not any column is used. Please use the model's search columns still as a fallback.